gcloud/logging/_gax.py

```diff
@@ -0,0 +1,252 @@
+# limitations under the License.
+
+"""GAX wrapper for Logging API requests."""
+
+import json
+
+# pylint: disable=import-error
+from google.gax import CallOptions
+from google.gax import INITIAL_PAGE
+from google.logging.type.log_severity_pb2 import LogSeverity
+from google.logging.v2.log_entry_pb2 import LogEntry
+from google.protobuf.json_format import Parse
+# pylint: enable=import-error
+
+from gcloud._helpers import _datetime_to_pb_timestamp
+
+
+class _LoggingAPI(object):
+    """Helper mapping logging-related APIs.
+
+    :type gax_api: :class:`google.logging.v2.logging_api.LoggingApi`
+    :param gax_api: API object used to make GAX requests.
+    """
+    def __init__(self, gax_api):
+        self._gax_api = gax_api
+
+    def list_entries(self, projects, filter_='', order_by='',
+                     page_size=0, page_token=None):
+        """Return a page of log entry resources.
+
+        :type projects: list of strings
+        :param projects: project IDs to include. If not passed,
+                         defaults to the project bound to the API's client.
+
+        :type filter_: str
+        :param filter_: a filter expression. See:
+                        https://cloud.google.com/logging/docs/view/advanced_filters
+
+        :type order_by: str
+        :param order_by: One of :data:`gcloud.logging.ASCENDING` or
+                         :data:`gcloud.logging.DESCENDING`.
+
+        :type page_size: int
+        :param page_size: maximum number of entries to return, If not passed,
+                          defaults to a value set by the API.
+
+        :type page_token: str
+        :param page_token: opaque marker for the next "page" of entries. If not
+                           passed, the API will return the first page of
+                           entries.
+
+        :rtype: tuple, (list, str)
+        :returns: list of mappings, plus a "next page token" string:
+                  if not None, indicates that more entries can be retrieved
+                  with another call (pass that value as ``page_token``).
+        """
+        options = _build_paging_options(page_token)
+        page_iter = self._gax_api.list_log_entries(
+            projects, filter_, order_by, page_size, options)
+        entries = [_log_entry_pb_to_mapping(entry_pb)
+                   for entry_pb in page_iter.next()]
+        token = page_iter.page_token or None
+        return entries, token
+
+    def write_entries(self, entries, logger_name=None, resource=None,
+                      labels=None):
+        """API call:  log an entry resource via a POST request
+
+        :type entries: sequence of mapping
+        :param entries: the log entry resources to log.
+
+        :type logger_name: string
+        :param logger_name: name of default logger to which to log the entries;
+                            individual entries may override.
+
+        :type resource: mapping
```

Comment:
Right. We don't require a specific type, only that it is duck-typable as a mapping.